cropharvest/eo/eo.py

```diff
@@ -231,34 +248,50 @@ def _bounding_box_from_centre(
         max_lat, min_lat = mid_lat + deg_lat, mid_lat - deg_lat
         max_lon, min_lon = mid_lon + deg_lon, mid_lon - deg_lon
 
-        return ee.Geometry.Polygon(
+        return (min_lon, min_lat, max_lon, max_lat), ee.Geometry.Polygon(
             [[[min_lon, min_lat], [min_lon, max_lat], [max_lon, max_lat], [max_lon, min_lat]]]
         )
 
+    @classmethod
     def _labels_to_polygons_and_years(
-        self, labels: geopandas.GeoDataFrame, surrounding_metres: int
-    ) -> List[Tuple[ee.Geometry.Polygon, str, int]]:
+        cls, labels: geopandas.GeoDataFrame, surrounding_metres: int
+    ) -> List[Tuple[ee.Geometry.Polygon, str, date, date]]:
 
         output: List[ee.Geometry.Polygon] = []
 
         print(f"Exporting {len(labels)} labels")
 
         for _, row in tqdm(labels.iterrows()):
+            latlons, bounding_box = cls._bounding_box_from_centre(
+                mid_lat=row["lat"], mid_lon=row["lon"], surrounding_metres=surrounding_metres
+            )
+
+            try:
+                export_identifier = row["export_identifier"]
+            except KeyError:
+                export_identifier = cls.make_identifier(
+                    latlons, row["start_date"], row["end_date"]
+                )
 
             output.append(
                 (
-                    self._bounding_box_from_centre(
-                        mid_lat=row["lat"],
-                        mid_lon=row["lon"],
-                        surrounding_metres=surrounding_metres,
-                    ),
-                    f"{row['index']}-{row['dataset']}",
-                    int(row["export_end_date"].year),
+                    bounding_box,
+                    export_identifier,
```

Comment:
To identify the export task inside Earth Engine I use the ee task description, unfortunately it can't just be the export_identifier due to a maximum length and characters, so somewhere around here we need to also add a description similar to: https://github.com/nasaharvest/crop-mask/blob/831d020d38a794afb07f8d270ab79ed6ad603232/src/ETL/ee_exporter.py#L222

Reply:
I'll do this in the next PR, which deals with the earth engine side of stuff (checking the ee queue for a task)